analysis_driver/driver.py

```diff
@@ -1,71 +1,58 @@
 
-logging_helper = util.NamedAppLogger('driver')
 
-
-def main(input_run_folder):
+def pipeline(input_run_folder):
     """
-    :param str input_run_folder: Absolute path to the input data directory, e.g. /abs/path/to/INPUT_DATA/run_id
+    :param str input_run_folder: Full path to an input data directory
     :return: Exit status
     :rtype: int
     """
-
-    # TODO: take all args, pass them to sub-functions in driver.py
-    # def analysis_driver():
-    #     run all the things
-    #
-    # def main():
-    #     validate arguments and pass them on
+    main_logger = logging.getLogger('main')
 
     run_id = os.path.basename(input_run_folder)
-    fastq_dir = os.path.join(config['fastq_dir'], run_id)
-    job_dir = os.path.join(config['jobs_dir'], run_id)
-    
-    logging.config.dictConfig(config.logging_config())
-    main_logger = util.NamedAppLogger('main')
+    fastq_dir = os.path.join(cfg['fastq_dir'], run_id)
+    job_dir = os.path.join(cfg['jobs_dir'], run_id)
+
+    main_logger.info('Input run folder (bcl data source): ' + input_run_folder)
+    main_logger.info('Fastq dir: ' + fastq_dir)
+    main_logger.info('Job dir: ' + job_dir)
 
     setup_working_dirs(fastq_dir, job_dir)
-    main_logger.info('Reading bcl data from ' + input_run_folder)
-    main_logger.info('Fastq path is ' + fastq_dir)
-
-    # Read RunInfo.xml and SampleSheet.csv for barcodes and masks
-    main_logger.info('Reading the reads info from ' + input_run_folder)
-
-    sample_sheet = analysis_driver.reader.sample_sheet.SampleSheet(input_run_folder)
-    run_info = analysis_driver.reader.run_info.RunInfo(input_run_folder)
-    if run_info.barcode_len:
-        if not sample_sheet.check_barcodes() == run_info.barcode_len:
-            main_logger.warn('Barcode mismatch: %s (SampleSheet.csv) and %s (RunInfo.xml)' % (
-                sample_sheet.check_barcodes(), run_info.barcode_len
-                )
-            )
-    else:
-        main_logger.warn('No barcode in RunInfo.xml')
+
+    sample_sheet = reader.sample_sheet.SampleSheet(input_run_folder)
+    run_info = reader.run_info.RunInfo(input_run_folder)
+    validate_run_info(run_info, sample_sheet)
 
     mask = run_info.mask.tostring(sample_sheet.check_barcodes())
     main_logger.info('bcl2fastq mask: ' + mask)  # example_mask = 'y150n,i6,y150n'
 
-    if config['job_execution'] == 'pbs':
+    if cfg['job_execution'] == 'pbs':
         run_pbs(
             logger=main_logger, input_run_folder=input_run_folder, job_dir=job_dir,
             run_id=run_id, fastq_dir=fastq_dir, mask=mask, sample_sheet=sample_sheet
         )
 
-    util.demultiplex_feedback(run_id)
+    # util.demultiplex_feedback(run_id)
 
     main_logger.info('Done')
     return 0
 
 
+def validate_run_info(run_info, sample_sheet):
+    assert run_info.barcode_len, 'No barcode found in RunInfo.xml'
+    assert sample_sheet.check_barcodes() == run_info.barcode_len, 'Barcode mismatch: %s (SampleSheet.csv) and %s (RunInfo.xml)' %\
```

Comment:
This should only raise a warning.
It is possible that the barcode in the samplesheet if smaller than the barcode in RunInfo.xml